.github/actions/e2e-test/action.yml

```diff
@@ -53,6 +53,11 @@ runs:
     - name: Run E2E Test
       shell: bash
       run: |
+        echo "::group::Install SNAPSHOT apm-application-toolkit"
+        ./mvnw -DskipTests -N install
+        ./mvnw -f apm-application-toolkit -DskipTests -am install
+        echo "::endgroup::"
```

Comment:
This is not enough, because after the 8.5.0, all related tests will fail again :)

I think once the e2e passed once, we could revert this and changes of e2e case back to normal.

Reply:
> This is not enough, because after the 8.5.0, all related tests will fail again :)
> 
> I think once the e2e passed once, we could revert this and changes of e2e case back to normal.

I consider the property `sw.version` in e2e pom.xml should be updated together when starting the next iteration

Reply:
How? Our release process doesn't do this automatically.

Reply:
> How? Our release process doesn't do this automatically.

Automation may be not a restriction IMO because there is only [one place](https://github.com/apache/skywalking/blob/6c6a8a7109c9a5a0dc173d8424ce45dec1da9f9e/test/e2e/e2e-service-provider/pom.xml#L37) to update. Also, forgetting to update it can be discovered (fail the CI) early when PR is opened, instead of failing someday after the newer version is released (we had this before), also we don't bother to release SNAPSHOTs just for testing. WDYT?

Reply:
If automation is the only concern, I may make it into the release automatical process later

Reply:
The concern is the process, snapshot->release version is automatically push and tag. Then, we will face the test will fail in official release tag.

Reply:
> Then, we will face the test will fail in official release tag.

Today we are already facing this case, [example](https://github.com/apache/skywalking/pull/6380). I just can see any downside of this PR but benefits I mentioned above

Reply:
I know we faced, that is why I don't want it happens again. As we are not real testing this feature in E2E(format I mean), we don't have to risk this again

Reply:
> that is why I don't want it happens again.

There is no guarantee that it won't happened again unless we decided not to do SNAPSHOT releases for testing. Also, using a previous version (8.4.0) in PR tests makes no sense to me, even if we don't test format here, it acts as a regression test that it doesn't break the non-format logs.

Reply:
Try this in the root, you will find the thing we are all happy

> printf 'VERSION=${project.version}\n0\n' | mvn org.apache.maven.plugins:maven-help-plugin:2.1.1:evaluate | grep '^VERSION'